### PR TITLE
Faster muon reconstruction using numba

### PR DESCRIPTION
**Pull request by RuneDominik**
Small changes to utilize numba for faster muon reconstruction. The major part in the ~1/3 decreased time required for processing muon events is gained in the direct implementation of a gaussian cdf, where the math modules erf-function is needed since standard numba is not aware of scipy. Further speedup is gained from decorating the chord_length function. This does, however, require the definition of an input and output layout that restricts the usage of the function to inputs of dimension 1 (which should be the use-case). As a result, the corresponding unit-tests are failing, as these test the function with scalar values.

### CODE REVIEW

File: ctapipe/image/muon/intensity_fitter.py [hunk 1]
```diff
@@ -194,6 +191,10 @@ def image_prediction(
         max_lambda_m=max_lambda.to_value(u.m),
     )
 
+# Define numba vectorized gaussian cdf
```
Comment (maxnoe) [inline ctapipe/image/muon/intensity_fitter.py:194]: Remove this and add a proper docstring to the function.

See: https://numpydoc.readthedocs.io/en/latest/format.html

File: ctapipe/image/muon/intensity_fitter.py [hunk 2]
```diff
@@ -194,6 +191,10 @@ def image_prediction(
...
+@vectorize([double(double, double, double)])
+def gaussian_cdf(x, mu, sig):
+    return 0.5*(1 + erf((x-mu)/(np.sqrt(2*sig**2))))
```
Comment (maxnoe) [inline ctapipe/image/muon/intensity_fitter.py:197]: Please use the black formatting pre-commit hook, see https://cta-observatory.github.io/ctapipe/getting_started/index.html

File: ctapipe/image/muon/intensity_fitter.py [hunk 3]
```diff
@@ -241,7 +242,9 @@ def image_prediction_no_units(
     # The weight is the integral of the ring's radial gaussian profile inside the
     # ring's width
     delta = pixel_diameter_rad / 2
-    cdfs = norm.cdf(
+
+    # Define Gaussian cdf for numba usage
```
Comment (maxnoe) [inline ctapipe/image/muon/intensity_fitter.py:246]: This comment does not really give more information since the function is aptly called `gaussian_cdf`

**Review by maxnoe — CHANGES_REQUESTED** [review]

**Comment by RuneDominik** [comment]
Included @maxnoe requests and swapped guvectorization to normal vectorization, as it includes scalar inputs without performance loss.

**Comment by kosack** [comment]
The pipelines are failing for another reason, see #1585   They should work once that is merged.

File: ctapipe/image/muon/intensity_fitter.py [hunk 4]
```diff
@@ -195,6 +196,28 @@ def image_prediction(
...
+    cumulative distribution function (cdf)
+
+    Parameters
+    ----------
+    x: float
+        point, at which the cdf should be evaluated
+    mu: float
+        gaussian mean
+    sig: float
+        gaussian standart deviation
```
Comment (maxnoe) [inline ctapipe/image/muon/intensity_fitter.py:212]: typo standard.

Comment (maxnoe) [reply ctapipe/image/muon/intensity_fitter.py:212]: Also with `vectorize` all values can also be `ndarrays`

File: ctapipe/image/muon/intensity_fitter.py [hunk 5]
```diff
@@ -195,6 +196,28 @@ def image_prediction(
...
+
+    Returns
+    -------
+    float: cdf-value at x
+    """
+    return 0.5 * (1 + erf((x - mu) / (np.sqrt(2 * sig ** 2))))
```
Comment (maxnoe) [inline ctapipe/image/muon/intensity_fitter.py:218]: sqrt is a pretty slow operation. sig is positive. So maybe we can gain something by defining a constant `sqrt2` and removing the sqrt here?

**Review by maxnoe — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**